Adding linter

lucie-pl opened the pull request:
Adding Rubocop, Brakeman and Reek gem to the project.

TODO : configure Docker

lucie-pl committed:
Adding rubocop gem

lucie-pl committed:
Adding brakeman gem

lucie-pl committed:
Adding reek gem

lucie-pl committed:
Running rubocop fixes

lucie-pl committed:
Fixing seeds after rubocop suggestions

lucie-pl committed:
Running reek fixes

lucie-pl committed:
Revert "Running rubocop fixes"

This reverts commit 367cc9c3733a798cce72d54b00ab0bfa49d6555d.

lucie-pl committed:
Running rubocop fixes: double quotes into single quotes

lucie-pl committed:
Running rubocop fixes: correcting <Missing top-level documentation> error in .rubocop.yml

quetzai commented on Gemfile:
```diff
@@ -38,38 +37,41 @@ gem "jbuilder"
 gem 'cocooned'
 gem 'devise'
+gem 'reek'
+gem 'rubocop', require: false
```
why is rubocop not required, but the 2 others are?

Interesting: https://blog.saeloun.com/2022/08/16/understanding_gemfile_and_gemfile_lock/#what-does-require-false-mean

quetzai replied:
the 3 are linters/code formatters/code checker.
Maybe they should be on `test` group or `dev` group rather than listed as a dependency needed for running this project.
https://blog.saeloun.com/2022/08/16/understanding_gemfile_and_gemfile_lock/#how-to-install-gems-in-specific-environments

quetzai reviewed (commented):
tiny change to have those 3 gems added in the right group.
optional: remove the commit adding rubocop changes and the commit that reverts it

lucie-pl committed:
Fixes after comment and review

- putting rubocop/reek/brakeman in test+dev group in Gemfile
- adding 'require, false' to load gems only when needed
- running rubocop fixes
